.github/workflows/docker-build-and-test.yml

```diff
@@ -1,21 +1,28 @@
 name: Docker Build and Test
 
 on:
-  workflow_dispatch:
-    inputs:
-      run_tests:
-        description: 'Run tests'
-        required: false
+  # Runs at every push on main
   push:
-    branches: [ "improvement/docker-testing-workflow-branch" ]  # Isolated branch name for testing to ensure we don't affect main
+    branches:
+      - main
+
+  # Runs at every pull request review
+  pull_request_review:
+    types: [submitted]
```

Comment:
What happens if after the review has been submitted another push is made to the open PR?
We would have potentially breaking changes, that are not covered by this test.

Reply:
Any possible problems will be caught right after merge, as every push to main triggers tests.